pages/api/users.ts

```diff
@@ -51,13 +59,22 @@ const handler = async (
   res: NextApiResponse,
   { session }: MiddlewareProps
 ): Promise<void> => {
-  switch (req.method) {
-    case "PUT":
-      await updateUserAdminStatus(req, res, session);
-      break;
-    case "GET":
-      await getUserAdminStatus(res);
-      break;
+  try {
+    if (!session) return res.status(401).json({ error: "Unauthorized" });
```

Comment:
On the type MiddlewareProps session may or may not exist depending on which middleware is run.  I agree that we know middleware runs and session will indeed be there but TypeScript will still infer that `session` can be undefined given it's optional status on MiddlewareProps.
If ever the sessionExists middleware is removed or modified by so that session is not included the check will ensure a graceful error and return instead of throwing a Type Error.